doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -4,99 +4,82 @@
-   the list of tactics associated to the head symbol of the goal and
-   tries to apply one of them (starting from the tactics with lower
-   cost). This process is recursively applied to the generated subgoals.
+   .. insertprodn auto_using hintbases
 
-   By default, :tacn:`auto` only uses the hypotheses of the current goal and
-   the hints of the database named ``core``.
+   .. prodn::
+      auto_using ::= using {+, @one_term }
+      hintbases ::= with *
+      | with {+ @ident }
 
-   .. warning::
+   :n:`@int_or_var`
+     Specifies the maximum search depth.  The default is 5.
 
-      :tacn:`auto` uses a weaker version of :tacn:`apply` that is closer to
-      :tacn:`simple apply` so it is expected that sometimes :tacn:`auto` will
-      fail even if applying manually one of the hints would succeed.
+   :n:`using {+, @one_term }`
 
-   .. tacv:: auto @natural
+      Uses lemmas :n:`{+, @one_term }` in addition to hints. If :n:`@one_term` is an
+      inductive type, the collection of its constructors are added as hints.
 
-      Forces the search depth to be :token:`natural`. The maximal search depth
-      is 5 by default.
+      Note that hints passed through the `using` clause are used in the same
+      way as if they were passed through a hint database. Consequently,
+      they use a weaker version of :tacn:`apply` and :n:`auto using @one_term`
+      may fail where :n:`apply @one_term` succeeds.
 
-   .. tacv:: auto with {+ @ident}
+      .. todo
+         Given that this can be seen as counter-intuitive, it could be useful
+         to have an option to use full-blown :tacn:`apply` for lemmas passed
+         through the `using` clause. Contributions welcome!
 
-      Uses the hint databases :n:`{+ @ident}` in addition to the database ``core``.
+   :n:`with *`
+     Use all existing hint databases. Using this variant is highly discouraged
+     in finished scripts since it is both slower and less robust than explicitly
+     selecting the required databases.
 
-      .. note::
+   :n:`with {+ @ident }`
+     Use the hint databases :n:`{+ @ident}` in addition to the database ``core``.
+     Use the fake database `nocore` to omit `core`.
 
-         Use the fake database `nocore` if you want to *not* use the `core`
-         database.
+   Implements a Prolog-like resolution procedure to solve the
+   current goal. It first tries to solve the goal using the :tacn:`assumption`
+   tactic, then it reduces the goal to an atomic one using :tacn:`intros` and
+   introduces the newly generated hypotheses as hints. Then it looks at
+   the list of tactics associated with the head symbol of the goal and
+   tries to apply one of them.  Lower cost tactics are tried before higher-cost
+   tactics.  This process is recursively applied to the generated subgoals.
 
-   .. tacv:: auto with *
+   If no `with` clause is givesn, :tacn:`auto` only uses the hypotheses of the
+   current goal and the hints of the database named ``core``.
```

Comment:
Typo: "givesn"

This sentence could be merged into the description of `with {+ @ident }`.